tools/resources.py

```diff
@@ -56,7 +57,13 @@ def _plugin_path_dependency() -> str:
             top_level_name, *_ = module_name.split(".", maxsplit=1)
             top_level_module = sys.modules.get(top_level_name)
             if top_level_module is not None:
-                top_level_directory = Path(inspect.getfile(top_level_module)).parent
+                try:
+                    top_level_directory = Path(inspect.getfile(top_level_module)).parent
+                except TypeError:
+                    # TypeError gets thrown if
+                    # pytest-xdist is used in tests since it
+                    # adds <module '__main__' (built-in)> into stack
```

Comment:
This still doesn't tell why it is **generally** safe to ignore the error. I wouldn't limit this only to pytest-xdist case. I would go something like:

```suggestion
                    # The 'getfile' function throws a TypeError for built-in modules.
                    # We are only interested in custom modules, so it is safe to ignore
                    # the error.
```